src/modules/fancyzones/lib/FancyZonesData.h

```diff
@@ -59,18 +60,18 @@ class FancyZonesData
         return appZoneHistoryMap;
     }
 
-    bool AddDevice(const std::wstring& deviceId);
-    void CloneDeviceInfo(const std::wstring& source, const std::wstring& destination);
-    void UpdatePrimaryDesktopData(const std::wstring& desktopId);
-    void RemoveDeletedDesktops(const std::vector<std::wstring>& activeDesktops);
+    bool AddDevice(const FancyZonesDataTypes::DeviceIdData& deviceId);
+    void CloneDeviceInfo(const FancyZonesDataTypes::DeviceIdData& source, const FancyZonesDataTypes::DeviceIdData& destination);
+    void UpdatePrimaryDesktopData(const GUID& desktopId);
+    void RemoveDeletedDesktops(const std::vector<GUID>& activeDesktops);
 
-    bool IsAnotherWindowOfApplicationInstanceZoned(HWND window, const std::wstring_view& deviceId) const;
-    void UpdateProcessIdToHandleMap(HWND window, const std::wstring_view& deviceId);
-    std::vector<size_t> GetAppLastZoneIndexSet(HWND window, const std::wstring_view& deviceId, const std::wstring_view& zoneSetId) const;
-    bool RemoveAppLastZone(HWND window, const std::wstring_view& deviceId, const std::wstring_view& zoneSetId);
-    bool SetAppLastZones(HWND window, const std::wstring& deviceId, const std::wstring& zoneSetId, const std::vector<size_t>& zoneIndexSet);
+    bool IsAnotherWindowOfApplicationInstanceZoned(HWND window, const FancyZonesDataTypes::DeviceIdData& deviceId) const;
+    void UpdateProcessIdToHandleMap(HWND window, const FancyZonesDataTypes::DeviceIdData& deviceId);
+    std::vector<size_t> GetAppLastZoneIndexSet(HWND window, const FancyZonesDataTypes::DeviceIdData& deviceId, const std::wstring_view& zoneSetId) const;
```

Comment:
Maybe not really work that should fall under this PR, but I guess no specific reason in having `zoneSetId` passed around as `std::wstring` when it is also `GUID` ?

Reply:
Due to this we have a large number of `StringFromCLSID` calls around the code, which are just making things harder.

Reply:
I agree. I can do this as a follow-up PR